tests/e2e/specs/auth-client-configuration.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { activatePlugin, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+describe( 'Providing client configuration', () => {
+
+	beforeEach( async() => {
+		await activatePlugin( 'google-site-kit' );
+		await visitAdminPage( 'admin.php', 'page=googlesitekit-splash' );
+	} );
+
+	it( 'Should have disabled button on load', async() => {
+
+		await page.waitForSelector( '#wizard-step-one-proceed' );
+
+		const proceedButton = await page.$$eval( '#wizard-step-one-proceed', el => el.map( x => x.hasAttribute( 'disabled' ) ) );
+
+		expect( proceedButton.length ).not.toEqual( 0 );
+	} );
+
+
+	it( 'Should have disabled button and display error when input is invalid', async() => {
+
+		await page.waitForSelector( '#client-configuration' );
+		page.click( '#client-configuration' );
+		await page.keyboard.type( 'This is not valid JSON' );
+
+		await page.waitForSelector( '.googlesitekit-error-text' );
+		const errorMessage = await page.$x(
+			'//p[contains(@class,"googlesitekit-error-text") and contains(text(), "Unable to parse client configuration values.")]'
+		);
+		expect( errorMessage.length ).not.toEqual( 0 );
+
+		const proceedButton = await page.$$eval( '#wizard-step-one-proceed', el => el.map( x => x.hasAttribute( 'disabled' ) ) );
+		expect( proceedButton.length ).not.toEqual( 0 );
+	} );
+
+	it( 'Should have enabled button valid value', async() => {
+
+		await page.waitForSelector( '#client-configuration' );
+		page.click( '#client-configuration' );
+
+		await page.waitFor( 1000 );
```

Comment:
Why the wait?

If there's something happening after the click, it's better to make this explicit, e.g. by using `waitForSelector`:

```js
await page.waitForSelector( '.my-awesome-element' );
```

Reply:
The first few chars of the string was not being included in the keyboard.type and the pause fixed. I'm not waiting for the focused class and working better.